### PR TITLE
Docusaurus v3

### PR DESCRIPTION
**Pull request by alvinometric**
- Upgrades to latest Docusaurus
- Upgrades our OpenAPI plugin to latest too
- Removes old, unneeded dependencies
- Our configs and sidebar files now use TypeScript

[Preview link](https://unleash-docs-git-alvin-docusaurus-v3-unleash-team.vercel.app/)

### CODE REVIEW

File: website/docs/how-to/how-to-add-sso-azure-saml.mdx [hunk 1]
```diff
@@ -104,6 +104,6 @@ Microsoft Entra limits the number of groups emitted in a SAML response to 150, i
 :::
 
 To enable group syncing in Unleash, do the following:
-1. In the Unleash Admin UI, go to **Admin > Single sign-on > SAML 2.0**.
+1. In the Unleash Admin UI, go to **Admin → Single sign-on → SAML 2.0**.
```
Comment (melindafekete) [inline website/docs/how-to/how-to-add-sso-azure-saml.mdx:107]:
```suggestion
1. In the Unleash Admin UI, go to **Admin > Single sign-on > SAML 2.0**.
```

File: website/docs/how-to/how-to-add-sso-azure-saml.mdx [hunk 2]
```diff
@@ -76,7 +76,7 @@ To find your Microsoft Entra identifier, go to the single sign-on settings of yo
 
 To finalize the configuration, do the following:
 
-1. In the Unleash Admin UI, go to **Admin > Single sign-on> SAML 2.0**.
+1. In the Unleash Admin UI, go to **Admin → Single sign-on → SAML 2.0**.
```
Comment (melindafekete) [inline website/docs/how-to/how-to-add-sso-azure-saml.mdx:79]:
```suggestion
1. In the Unleash Admin UI, go to **Admin > Single sign-on > SAML 2.0**.
```

File: website/docs/how-to/how-to-add-sso-azure-saml.mdx [hunk 3]
```diff
@@ -60,7 +60,7 @@ Save the following information from the single sign-on settings of your applicat
 - [Microsoft Entra Identifier](#microsoft-entra-identifier)
 
 #### SAML certificate
-To save the SAML certificate, go to the single sign-on settings of your application. In **SAML Certificates > Federation Metadata XML**, click **Download**. Open the file and copy the contents between the `X509Certificate` tag. 
+To save the SAML certificate, go to the single sign-on settings of your application. In **SAML Certificates → Federation Metadata XML**, click **Download**. Open the file and copy the contents between the `X509Certificate` tag.
```
Comment (melindafekete) [inline website/docs/how-to/how-to-add-sso-azure-saml.mdx:63]:
```suggestion
To save the SAML certificate, go to the single sign-on settings of your application. In **SAML Certificates > Federation Metadata XML**, click **Download**. Open the file and copy the contents between the `X509Certificate` tag. 
```

File: website/docs/reference/network-view.png
[the remark this answers is not in the document]
Comment (alvinometric) [reply website/docs/reference/network-view.png:1]: Thank you 🙏 Do we need the lines to extend that far out?

[the remark this answers is not in the document]
Comment (melindafekete) [reply website/docs/reference/network-view.png:1]: Made the lines a bit shorter, but still want it to look like a sequence diagram

Screenshot 2024-10-29 at 09 46 36

[the remark this answers is not in the document]
Comment (melindafekete) [reply website/docs/reference/network-view.png:1]: Made them a bit shorter, but still sequence diagrammy a4f1e2e6382658116e4ffbb5a74da2543d0ae607

**Review by melindafekete — APPROVED** [review]
Best day of my life ❤️

File: website/clean-generated-docs.mjs [hunk 4]
```diff
@@ -15,7 +15,7 @@
 // the bits that are specific to the generation source we use, and make the docs
 // easier to use. In particular, removing the leading `/ushosted` is likely to
 // save us loooots of questions.
-const replace = require('replace-in-file');
+import { replaceInFileSync } from 'replace-in-file';
```
Comment (melindafekete) [inline website/clean-generated-docs.mjs:18]: Here

**Pull request merged**